src/CommunityToolkit.Aspire.Hosting.PowerShell/PowerShellRunspacePoolResourceBuilderExtensions.cs

```diff
@@ -0,0 +1,124 @@
+
+/// <summary>
+/// Extensions for the PowerShellRunspacePoolResourceBuilder.
+/// </summary>
+public static class PowerShellRunspacePoolResourceBuilderExtensions
+{
+    /// <summary>
+    /// Adds a PowerShell script resource to the distributed application.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="name"></param>
+    /// <param name="script"></param>
+    /// <returns></returns>
+    public static IResourceBuilder<PowerShellScriptResource> AddScript(
+        this IResourceBuilder<PowerShellRunspacePoolResource> builder,
+        [ResourceName] string name,
+        [StringSyntax("PowerShell")] string script)
+    {
+        ArgumentException.ThrowIfNullOrWhiteSpace(name);
+        ArgumentException.ThrowIfNullOrWhiteSpace(script);
+
+        // parse to force an early exception if the script is invalid
+        var scriptBlock = ScriptBlock.Create(script);
+
+        var scriptResource = new PowerShellScriptResource(name, scriptBlock, builder.Resource);
+
+        builder.ApplicationBuilder.Eventing.Subscribe<InitializeResourceEvent>(scriptResource, (e, ct) =>
+        {
+            var loggerService = e.Services.GetRequiredService<ResourceLoggerService>();
+            var notificationService = e.Services.GetRequiredService<ResourceNotificationService>();
+
+            var scriptName = scriptResource.Name;
+            var scriptLogger = loggerService.GetLogger(scriptName);
+            try
+            {
+                // TODO: capture script streams and log them
+                scriptLogger.LogInformation("Starting script '{ScriptName}'", scriptName);
+
+                _ = notificationService
+                    .WaitForDependenciesAsync(scriptResource, ct)
+                    .ContinueWith(
+                        async (_) => await scriptResource.StartAsync(scriptLogger, notificationService, ct),
+                        ct);
+            }
+            catch (Exception ex)
+            {
+                scriptLogger.LogError(ex, "Failed to start script '{ScriptName}'", scriptName);
+            }
+
+            return Task.CompletedTask;
+        });
+
+        return builder.ApplicationBuilder
+            .AddResource(scriptResource)
+            .WaitFor(builder) // wait for pool resource
+            .WithParentRelationship(builder.Resource) // owned by pool
+            .WithInitialState(new()
+            {
+                ResourceType = "PowerShellScript",
+                State = KnownResourceStates.NotStarted,
+                Properties = [
+                    new ("Script", script),
+                    new("RunspacePool", builder.Resource.Name)
+                ]
+            })
+            .ExcludeFromManifest()
+            .WithCommand("break", "Stop script execution",
+            async _ =>
+            {
+                await scriptResource.BreakAsync();
+                return CommandResults.Success();
+            },
+            new CommandOptions
```

Comment:
Would it be worthwhile having a command to rerun the script too?

Reply:
I did think about it, but there are some things to be aware of - namely that a runspace pool is somewhat stateful in terms of loaded modules, asssemblies etc. In 99.9% of cases this won't matter. That said, it might require a bit of a rearchitecture should we decide to add that feature (which I would like, tbh)

https://github.com/PowerShell/PowerShell/issues/7131